test/unit/product/product-view.js

```diff
@@ -54,40 +55,210 @@ describe('Product View class', () => {
+      wrapper.appendChild(node);
+      product.view.wrapper = wrapper;
+    });
+
+    afterEach(() => {
+      resizeStub.restore();
+      addEventListenerStub.restore();
+    });
+
+    it('does not add event listener if product contents do not have an image or image with carousel', () => {
+      product.config.product.contents = {
+        img: null,
+        imgWithCarousel: null,
+      };
+      product.view.resizeOnLoad();
+      assert.notCalled(addEventListenerStub);
+    });
+
+    it('does not add event listener if there is no image in the wrapper', () => {
+      product.config.product.contents = {
+        img: {id: '123'},
+        imgWithCarousel: null,
+      };
+      product.view.resizeOnLoad();
+      assert.notCalled(addEventListenerStub);
+    });
+
+    it('adds resize event listener on load if product contents have an image and there is an image in the wrapper', () => {
+      node.className += 'img-class';
+      product.config.product.contents = {
+        img: {id: '123'},
+        imgWithCarousel: null,
+      };
+
+      product.view.resizeOnLoad();
+      assert.calledOnce(addEventListenerStub);
+      assert.calledWith(addEventListenerStub, 'load', sinon.match.func);
+      addEventListenerStub.getCall(0).args[1]();
+      assert.calledOnce(resizeStub);
+    });
+
+    it('adds resize event listener on load if product contents have an image with a carousel and there is an image in the wrapper', () => {
+      node.className += 'img-class';
+      product.config.product.contents = {
+        img: null,
+        imgWithCarousel: {id: '123'},
+      };
+
+      product.view.resizeOnLoad();
+      assert.calledOnce(addEventListenerStub);
+      assert.calledWith(addEventListenerStub, 'load', sinon.match.func);
+      addEventListenerStub.getCall(0).args[1]();
+      assert.calledOnce(resizeStub);
+    });
+  });
+
+  describe('render', () => {
+    let superRenderStub;
+    let resizeOnLoadStub;
+
+    beforeEach(() => {
+      superRenderStub = sinon.stub(View.prototype, 'render');
+      resizeOnLoadStub = sinon.stub(product.view, 'resizeOnLoad');
+    });
+
+    afterEach(() => {
+      superRenderStub.restore();
+      resizeOnLoadStub.restore();
+    });
+
+    it('calls super render', () => {
+      product.view.render();
+      assert.calledOnce(superRenderStub);
+    });
+
+    it('resizes on load', () => {
+      product.view.render();
+      assert.calledOnce(resizeOnLoadStub);
+    });
+  });
+
   describe('wrapTemplate()', () => {
     beforeEach(async () => {
       await product.init(testProductCopy);
+      product = Object.defineProperty(product, 'isButton', {
+        writable: true,
+      });
     });
 
-    describe('when isButton() is false', () => {
-      it('calls super', () => {
+    it('wraps html in a div with wrapper class and product class if the component is not a button', () => {
+      product.isButton = false;
+      const htmlString = product.view.wrapTemplate('test');
```

Comment:
For consistency with other tests, can we update this variable name to `string`, or update everything to `htmlString`?